components/exercises/chapter-4-oscar/AList/nth.tsx

```diff
@@ -0,0 +1,16 @@
+import { ReactNode } from 'react';
+import { List } from './CityList';
+
+/**
+ * Returns the value of the nth element in a list.
+ * @param list - The list to search for the nth element.
+ * @param n - The index of the element to retrieve.
+ * @returns The value of the nth element in the list, or undefined if the list is empty or n is out of bounds.
+ */
+
+export function nth(list: List | null, n: number): undefined | ReactNode {
```

Comment:
This shouldn't return `ReactNode`, but the value of `List` which is `string | number;`. 

FYI Typescript isn't complaining because [ReactNode includes, among other types, `string | number`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L244) . 

If possible, for cleanliness, let's keep react code on react files

Reply:
you are right, thank you